src/features/hotspots/root/HotspotsView.tsx

```diff
@@ -171,7 +174,9 @@ const HotspotsView = ({ ownedHotspots }: Props) => {
       >
         <Map
           ownedHotspots={ownedHotspots}
-          selectedHotspots={selectedHotspot ? [selectedHotspot] : undefined}
+          selectedHotspots={
+            selectedHotspot && hasLocation ? [selectedHotspot] : undefined
```

Comment:
I believe the `[selectedHotspot]` will be a different object on every render. Might want to memoize it.